frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripParentServlet.java

```diff
@@ -46,25 +46,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     logger.atInfo().log("received doPost");
     try {
-      Optional<TripModel> requestTrip =
-          ServletUtil.extractFromRequestBody(request, gson, TripModel.class);
-
-      if (requestTrip.isPresent()) {
-        try {
-          TripModel resolvedTrip = resolveDefaults(requestTrip.get());
-          storage.addTrip(resolvedTrip);
-          response.setContentType("application/json");
-          PrintWriter writer = response.getWriter();
-          writer.println(gson.toJson((resolvedTrip)));
-          writer.flush();
-          writer.close();
-        } catch (TripAlreadyExistsException e) {
-          response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-        }
-
-      } else {
-        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-      }
+      TripModel requestTrip = ServletUtil.extractFromRequestBody(request.getReader(), gson, TripModel.class);
+      TripModel resolvedTrip = resolveDefaults(requestTrip);
+      storage.addTrip(resolvedTrip);
+      response.setContentType("application/json");
+      PrintWriter writer = response.getWriter();
+      writer.println(gson.toJson((resolvedTrip)));
+      writer.flush();
```

Comment:
FYI, you only need to call one of flush or close. Assuming it's safe to do so, I'd recommend calling `close()` and doing so in a finally block for good measure.